### PR TITLE
adds swipe to delete fav stop feature

### PR DESCRIPTION
**Pull request by toastjpg**
Changes:
- Adds a swipe handler to listen to left swipe actions on the fav stop cards
  - Fav stop card view has a red background + trash icon that is revealed on swipe
- Added _id field to the favourite stop data models
- Added specific adapter notifiers depending on the operation
  - Reduces the amount of `notifyDataSetChanged()` calls we make

### CODE REVIEW

File: app/src/main/java/ca/wheresthebus/MainDBViewModel.kt [hunk 1]
```diff
@@ -163,15 +141,7 @@ class MainDBViewModel : ViewModel() {
         ).find().take(10)
```
Comment (bryanjhdang) [inline app/src/main/java/ca/wheresthebus/MainDBViewModel.kt:141]: Unrelated to this PR, but if this is going to be a user setting, we should add it as a variable somewhere.

Comment (toastjpg) [reply app/src/main/java/ca/wheresthebus/MainDBViewModel.kt:141]: 👍

File: app/src/main/res/layout/item_fav_bus.xml [hunk 2]
```diff
@@ -1,36 +1,59 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+<FrameLayout
```
Comment (bryanjhdang) [inline app/src/main/res/layout/item_fav_bus.xml:2]: Is there a reason why this is a Frame instead of a CardView now?

**Review by bryanjhdang — APPROVED** [review]
Good work on adding ID.

File: app/src/main/res/layout/item_fav_bus.xml (hunk 2 above)
Comment (toastjpg) [reply app/src/main/res/layout/item_fav_bus.xml:2]: Using frame layout to display them on top of each other in a foreground/background style. The red delete color + trash icon is in a relative layout behind the card view.

**Pull request merged**